File: R/process/run_toy_data.py
import decoupler as dc
import pandas as pd

# Read data
mat = pd.read_csv('data/raw/toy_mat.csv', index_col=0).T
net = pd.read_csv('data/raw/toy_net.csv')
net.columns = ['source', 'target', 'weight']

# Args
seed = 42
n_jobs = 8
opts_list = {
  'udt' : {'min_leaf':3, 'seed':seed},
  'mdt' : {'trees':10, 'min_leaf':3, 'n_jobs':n_jobs, 'seed':seed},
  'aucell' : {'n_up':3, 'seed':seed},
  'wmean' : {'times':100, 'seed':seed},
  'wsum' : {'times':100, 'seed':seed},
  'ulm' : {},
  'mlm' : {},
  'viper' : {'pleiotropy':True},
  'gsva' : {'kcdf':True},
  'ora' : {'n_up':3, 'n_bottom':0, 'n_background':20000, 'seed':seed},
  'gsea' : {'times':100, 'seed':seed}
}

# Run
res = dc.decouple(mat, net, min_n=0, verbose=False, methods='all', args=opts_list)

# Melt and sort
res = dc.melt(res).sort_values(['method', 'source', 'sample'])

# Store
res.to_csv('data/prc/toy_acts_Python.csv', index=None)

File: R/process/run_toy_data.R
library(decoupleR)
library(tidyverse)

# Get prc path
raw_path <- file.path('data', 'raw')
prc_path <- file.path('data', 'prc')
dir.create(prc_path, showWarnings = F, recursive = T)

# Get toy data-set
lst <- decoupleR::get_toy_data()
mat <- lst$mat
network <- lst$network

write.csv(mat, file.path(raw_path, 'toy_mat.csv'))
write.csv(network, file.path(raw_path, 'toy_net.csv'), row.names = F)

# Args
seed = 42
nproc = 8
opts_list <- list(
  udt = list(sparse=FALSE, center=FALSE, na.rm=FALSE, min_n = 3, seed=seed),
  mdt = list(sparse=FALSE, center=FALSE, na.rm=FALSE, trees = 10, min_n = 3,
             nproc = nproc, seed=seed),
  aucell = list(nproc=nproc, seed=seed, aucMaxRank=3),
  wmean = list(times=100, sparse=FALSE, randomize_type = "rows", seed=seed),
  wsum = list(times=100, sparse=FALSE, randomize_type = "rows", seed=seed),
  ulm = list(sparse=FALSE, center=FALSE, na.rm=FALSE),
  mlm = list(sparse=FALSE, center=FALSE, na.rm=FALSE),
  viper = list(verbose=FALSE, pleiotropy=T, eset.filter=F),
  gsva = list(verbose = FALSE, method = "gsva"),
  ora = list(n_up=3, n_bottom=0, n_background=20000, with_ties = TRUE, seed=seed),
  fgsea = list(times=100, nproc=nproc, seed=seed)
)

# Run
res <- decoupleR::decouple(mat,
                           network,
                           args = opts_list,
                           statistics = 'all',
                           minsize=0) %>%
  select(-run_id)

# Rename
res$statistic[res$statistic=='corr_wmean'] <- 'wmean_corr'
res$statistic[res$statistic=='norm_wmean'] <- 'wmean_norm'
res$statistic[res$statistic=='corr_wsum'] <- 'wsum_corr'
res$statistic[res$statistic=='norm_wsum'] <- 'wsum_norm'
res$statistic[res$statistic=='fgsea'] <- 'gsea'
res$statistic[res$statistic=='norm_fgsea'] <- 'gsea_norm'
colnames(res) <- c("method", "source", "sample", "score", "pval")

# Sort, filter
res <- res %>%
  arrange(method, source, sample) %>%
  select(sample, source, score, method, pval) %>%
  mutate(score=ifelse(is.finite(score), score, NA))

# Store
write.csv(res, file.path(prc_path, 'toy_acts_R.csv'), row.names = F)
